### PR TITLE
Print help for -? and /? cross-plat

### LINKED ISSUE #42395: `build.cmd /?` at the root just dumps a list of subsets and errors.
`build.cmd /?` at the root just dumps a list of subsets and errors. 

It should behave the same as `build -?` `build --help` `build -h` which have the great helpful output with examples.

BTW it would be good to make `build.sh /?` work on Linux as well, as my muscle memory led me down that path and I had to debug the script before I realized that 10 mins later. I had a commit to fix that which I can dig up but it was a tiny change and maybe we can fix both here.

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/runtime/issues/42395

### CODE REVIEW

**Review by danmoseley — APPROVED** [review]
thanks @ViktorHofer

File: build.cmd [hunk 1]
```diff
@@ -3,6 +3,7 @@ setlocal
 
 set _args=%*
 if "%~1"=="-?" set _args=-help
+if "%~1"=="/?" set _args=-help
```
Comment (ivdiazsa) [inline build.cmd:6]: Do we also want `/help` and/or `/h` as flags? If so, then we'd need to add their own clauses:

```cmd
if "%~1"=="/help" set _args=-help
```

**Comment by ivdiazsa** [comment]
While trying to reproduce this issue, I found something interesting in `build.ps1`, in the following code snippet:

```powershell
if ($help -or (($null -ne $properties) -and ($properties.Contains('/help') -or $properties.Contains('/?')))) {
  Get-Help
  exit 0
}
```

The `$properties` is always empty in this clause, so we might as well remove those checks, and stay only with the `$help` one.

File: build.cmd (hunk 1 above)
Comment (ViktorHofer) [reply build.cmd:6]: I don't think we want any `/{letter}` abbreviations. I would only support the ones that were considered by @danmosemsft.

**Comment by ivdiazsa** [comment]
This was an accident. Pinged Dan to approve the PR again.

**Review by ivdiazsa — APPROVED** [review]

**Pull request merged**